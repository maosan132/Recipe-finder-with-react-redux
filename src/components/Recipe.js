import React from 'react';

const Recipe = () => (
  <div>
    A recipe
  </div>
);

export default Recipe;
