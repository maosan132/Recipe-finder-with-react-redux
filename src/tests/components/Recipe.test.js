import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Recipe from '../../components/Recipe';

describe('<Recipe />', () => {
  const recipe = {
    recipes: {
      meals: [{ 0: { name: 'Recipe1', category: 'Vegetarian' } },
        { 1: { name: 'Recipe2', category: 'Seafood' } }],
    },
  };

  it('it renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Recipe recipe={recipe} />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an image', () => {
    const { queryAllByTestId } = render(
      <Router>
        <Recipe recipe={recipe} />
      </Router>,
    );
    const image = queryAllByTestId('image');
    expect(image[0]).toBeTruthy();
  });

  it('renders a name', () => {
    const { queryAllByTestId } = render(
      <Router>
        <Recipe recipe={recipe} />
      </Router>,
    );
    const name = queryAllByTestId('name');
    expect(name[0]).toBeTruthy();
  });

  it('renders an image corecctly', () => {
    const { queryAllByTestId } = render(
      <Router>
        <Recipe recipe={recipe} />
      </Router>,
    );
    const image = queryAllByTestId('img');
    expect(image[0]).toBeFalsy();
  });

  it('renders a name', () => {
    const { queryAllByTestId } = render(
      <Router>
        <Recipe recipe={recipe} />
      </Router>,
    );
    const name = queryAllByTestId('name');
    expect(name[0]).not.toBeFalsy();
  });
});
