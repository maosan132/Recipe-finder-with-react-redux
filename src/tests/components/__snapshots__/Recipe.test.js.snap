// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipe /> it renders correctly 1`] = `
<div
  className="Card card"
>
  <div
    className="card-header"
  >
    <div
      className="title card-title h5"
      data-testid="name"
    />
  </div>
  <img
    className="card-img-top"
    data-testid="image"
  />
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      Category: undefined
    </p>
    <a
      href="/recipeInfo/undefined"
      onClick={[Function]}
      style={
        Object {
          "color": "#fff",
          "textDecoration": "none",
        }
      }
    >
      <button
        className="button btn btn-primary"
        disabled={false}
        type="button"
      >
        Recipe Details
      </button>
    </a>
  </div>
</div>
`;
